### PR TITLE
Report missing header files as check comments

### PR DESCRIPTION
**Pull request by florian-kuebler**
Test: CI

### CODE REVIEW

**Review by antonrohr — APPROVED** [review]
I guess lgtm!

**Review by beckerhe — APPROVED** [review]
Nice. Otherwise LGTM

File: contrib/scripts/report-license-headers.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+if [[ ! -s "${GITHUB_WORKSPACE}/missing_license_headers.txt" ]] ; then
+  exit 0
+fi
+
+format_annotations "${GITHUB_WORKSPACE}/missing_license_headers.txt" | curl \
+  -X POST \
+  -H "Accept: application/vnd.github+json" \
+  -H "Authorization: Bearer ${GITHUB_TOKEN}" \
+  https://api.github.com/repos/florian-kuebler/orbitprofiler/check-runs \
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply contrib/scripts/report-license-headers.sh:49]: Done.

File: contrib/scripts/report-license-headers.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  echo "\"title\":\"Missing license headers report\","
+  echo "\"summary\":\"You are missing the license headers in the following files.\","
+  echo "\"annotations\":["
+  FIRST_LINE=1
+  while read -r line; do
+    if [[ -z $FIRST_LINE ]]; then
+      echo ","
+    fi
+    FIRST_LINE=""
+    echo "{\"path\":\"$line\","
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply contrib/scripts/report-license-headers.sh:28]: Done.

File: contrib/scripts/report-license-headers.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+# Fail on any error.
+set -euo pipefail
+
+function format_annotations() {
+  local files_missing_license_headers=$1
+
+  echo "{"
+  echo "\"name\":\"missing-license-headers\","
+  echo "\"head_sha\":\"$COMMIT_SHA\","
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply contrib/scripts/report-license-headers.sh:15]: Done.

**Pull request merged**